### PR TITLE
Label ETLs support

### PR DESCRIPTION
**Pull request by OmerKorenAF**

### CODE REVIEW

File: anomaly_detection_src/dags/labels_etl.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    logging.info('Done uploading labels data')
+
+
+use_cases = ['rides_number', 'fallbacks', 'price_per_ride']
+
+for curr_use_case in use_cases:
+    dag_id = f'{curr_use_case}_anomaly_detection_labels_etl'
+    globals()[dag_id] = DAG(
+        dag_id=dag_id,
+        schedule_interval='0 0 * * *',
```
Comment (drorSalti) [inline anomaly_detection_src/dags/labels_etl.py:50]: why once a day and not every hour?

**Review by drorSalti — APPROVED** [review]

File: anomaly_detection_src/dags/labels_etl.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    labels_final_df.to_gbq(
+        f"{gcp_params['bq_schema']}.{gcp_params['bq_labels_table_name']}",
+        project_id=gcp_params['bq_project'],
+        if_exists='append'
+    )
+
+    logging.info('Done uploading labels data')
+
+
+def back_fill():
```
Comment (drorSalti) [inline anomaly_detection_src/dags/labels_etl.py:43]: why we need this?

File: anomaly_detection_src/common/data_getter.py [hunk 3]
```diff
@@ -27,36 +27,91 @@ def run_query(query, query_params=None):
...
         return None
     datetime_hours_df = pd.DataFrame({'datetime_hour': pd.date_range(start_date, end_date, freq='H')})
-    df = datetime_hours_df.merge(raw_data_df, on='datetime_hour', how='left').fillna(0).reset_index(drop=True)
-    df['time_id'] = df['datetime_hour'].apply(
-        lambda ts: from_datetime_hour_to_hour_bucket(ts, hour_buckets_size=hour_buckets_size)
+    labels_df = datetime_hours_df.merge(
+        labels_raw_data_df,
+        on='datetime_hour',
+        how='left'
+    ).fillna(0).reset_index(drop=True)
```
Comment (drorSalti) [inline anomaly_detection_src/common/data_getter.py:41]: fillna(0) is relevant for all kpis? I'm not sure

File: anomaly_detection_src/common/data_getter.py [hunk 4]
```diff
@@ -27,36 +27,91 @@ def run_query(query, query_params=None):
...
+    return labels_df
+
+
+def preprocess_labels_raw_data(labels_raw_df, start_ts, end_ts, agg_func):
+    date_range = pd.date_range(start=start_ts + timedelta(hours=24), end=end_ts, freq='H')
```
Comment (drorSalti) [inline anomaly_detection_src/common/data_getter.py:46]: it's a bit weird and confusing IMHO

**Review by drorSalti — APPROVED** [review]

File: anomaly_detection_src/common/data_getter.py (hunk 3 above)
Comment (OmerKorenAF) [reply anomaly_detection_src/common/data_getter.py]: You are right. I added default value support.

File: anomaly_detection_src/dags/labels_etl.py (hunk 2 above)
Comment (OmerKorenAF) [reply anomaly_detection_src/dags/labels_etl.py]: Deleted it.

**Pull request merged**